clippy_lints/src/copies.rs

```diff
@@ -40,6 +40,28 @@ declare_clippy_lint! {
     "consecutive `ifs` with the same condition"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for consecutive `if`s with the same function call.
+    ///
+    /// **Why is this bad?** This is probably a copy & paste error.
+    /// Despite the fact that function can have side effects and `if` works as
+    /// intended, such an approach is implicit and can be considered a "code smell".
+    ///
+    /// **Known problems:** Hopefully none.
+    ///
+    /// **Example:**
+    /// ```ignore
+    /// if foo() {
+    ///     …
+    /// } else if foo() {
+    ///     …
+    /// }
+    /// ```
+    pub IFS_SAME_COND_FN,
```

Comment:
`SAME_FUNCTIONS_IN_IF_CONDITION`, as per naming conventions. (putting `allow` before the lint name should produce an english sentence (more or less))